src/modules/reward/screens/RewardConfirm/index.tsx

```diff
@@ -1,63 +1,58 @@
 import { getAtokenInfo } from '../../../../helpers/get-atoken-info';
+
 import ScreenWrapper from '../../../../components/wrappers/ScreenWrapper';
 import ContentWrapper from '../../../../components/wrappers/ContentWrapper';
 import Row from '../../../../components/basic/Row';
 import PoolTxConfirmationView from '../../../../components/PoolTxConfirmationView';
 import Value from '../../../../components/basic/Value';
+import Link from '../../../../components/basic/Link';
 
 import messages from './messages';
 
 export function RewardConfirm() {
   const intl = useIntl();
-  const { incentiveService } = useTxBuilderContext();
-  const { userUnclaimedRewards } = useStaticPoolDataContext();
+  const location = useLocation();
+
   const { user, reserves } = useDynamicPoolDataContext();
-  const { networkConfig } = useProtocolDataContext();
+  const { userIncentives, txBuilder } = useIncentivesDataContext();
+  const incentivesControllerAddress = location.pathname.split('/')[3];
+  const incentiveData: UserIncentiveData = userIncentives[incentivesControllerAddress];
+  const rewardTokenSymbol = getRewardTokenSymbol(reserves, incentiveData.rewardTokenAddress);
 
   const aTokenData = getAtokenInfo({
-    address: incentiveService.incentivesControllerRewardTokenAddress,
-    symbol: networkConfig.rewardTokenSymbol, // TODO: maybe need change in the future
-    decimals: 18,
+    address: incentiveData.rewardTokenAddress,
+    symbol: rewardTokenSymbol,
+    decimals: incentiveData.rewardTokenDecimals,
     withFormattedSymbol: true,
   });
 
-  if (!user) return null;
-
-  const amount = valueToBigNumber(user.totalRewards).plus(userUnclaimedRewards);
-  if ((amount.lt(0) && !amount.eq(-1)) || !user) {
+  if ((incentiveData.claimableRewards.lt(0) && !incentiveData.claimableRewards.eq(-1)) || !user) {
     return null;
   }
 
   let blockingError = '';
-  if (amount.eq('0')) {
+  if (incentiveData.claimableRewards.eq('0')) {
     blockingError = intl.formatMessage(messages.notEnoughBalance);
   }
 
-  const formattedAmount = amount.toString();
-
-  const assets = user.reservesData.reduce((acc, userReserve) => {
-    const reserve = reserves.find(
-      (reserve) =>
-        reserve.underlyingAsset.toLowerCase() === userReserve.reserve.underlyingAsset.toLowerCase()
-    ) as ComputedReserveData;
-    if (userReserve.aTokenRewards !== '0') acc.push(reserve.aTokenAddress);
-    if (userReserve.vTokenRewards !== '0') acc.push(reserve.variableDebtTokenAddress);
-    if (userReserve.sTokenRewards !== '0') acc.push(reserve.stableDebtTokenAddress);
-    return acc;
-  }, [] as string[]);
+  const formattedAmount = normalize(
+    incentiveData.claimableRewards,
+    incentiveData.rewardTokenDecimals
+  );
 
+  const assets = incentiveData.assets;
   const handleGetTransactions = async () =>
-    incentiveService.claimRewards({ user: user.id, assets, to: user.id });
+    txBuilder.claimRewards({ user: user.id, assets, to: user.id });
```

Comment:
this should be inentiveService or similar, as txBuilder is a reserved name for the TxBuilder (and this is just a service passed down)